runtime/interpreter/decode.go

```diff
@@ -1807,15 +1807,15 @@ func (d TypeDecoder) decodeDictionaryStaticType() (StaticType, error) {
 	return NewDictionaryStaticType(d.memoryGauge, keyType, valueType), nil
 }
 
-func (d TypeDecoder) decodeRestrictedStaticType() (StaticType, error) {
-	const expectedLength = encodedRestrictedStaticTypeLength
+func (d TypeDecoder) decodeIntersectionStaticType() (StaticType, error) {
```

Comment:
I assume this and the other internal decoding functions stay backward compatible, i.e. they'll be able to decode data stored in the old format?

Reply:
For the ccf and interpreter decoders, yes I believe they remain backwards compatible, although it might be worth asking @fxamacker to double check this.

The JSON decoder will change because the old fields were named things like `"restrictions"` or the tag was "RestrictedType", which have been changed to have more appropriate names.

Reply:
> For the ccf and interpreter decoders, yes I believe they remain backwards compatible

Yes, CCF and also internal value encodings are backwards compatible.  :+1: